build.rs

```diff
@@ -170,147 +204,149 @@ fn probe_r_paths() -> io::Result<InstallationPaths> {
     let mut lines = out.lines();
 
-    let major = match lines.next() {
-        Some(line) => line.to_string(),
-        _ => return Err(EnvVarError::InvalidROutput("Cannot find R major version")),
+    // Process the first line of the output
+    let r_version = match lines.next() {
+        Some(v) => v.to_string(),
+        None => return Err(EnvVarError::InvalidROutput("Cannot find R version")),
     };
-
-    let minor = match lines.next() {
-        Some(line) => line.to_string(),
-        _ => return Err(EnvVarError::InvalidROutput("Cannot find R minor version")),
-    };
-
-    let patch = match lines.next() {
-        Some(line) => line.to_string(),
-        _ => return Err(EnvVarError::InvalidROutput("Cannot find R patch level")),
-    };
-
-    let devel = match lines.next() {
-        Some(line) => if line == "TRUE" {
-            "-devel".to_string()
-        } else {
-            "".to_string()
-        },
-        _ => return Err(EnvVarError::InvalidROutput("Cannot find R development status")),
-    };
-
     let version_string = match lines.next() {
         Some(line) => line.to_string(),
         _ => return Err(EnvVarError::InvalidROutput("Cannot find R version string")),
     };
 
-    Ok(RVersionInfo {
-        major,
-        minor,
-        patch,
-        devel,
-        version_string,
-    })
+    parse_r_version(r_version, Some(version_string))
 }
 
-fn get_r_version(r_version_env_var: &str, r_paths: &InstallationPaths) -> Result<RVersionInfo, EnvVarError> {
-    match get_r_version_from_env(r_version_env_var)
-    {
+fn get_r_version(
+    r_version_env_var: &str,
+    r_paths: &InstallationPaths,
+) -> Result<RVersionInfo, EnvVarError> {
+    // Try looking for the envvar first.
+    match get_r_version_from_env(r_version_env_var) {
+        // If the envvar is found and it can be parsed as a valid RVersionInfo, use it.
         Ok(v) => Ok(v),
-        Err(EnvVarError::InvalidEnvVar(e)) => Err(EnvVarError::InvalidEnvVar(e)),
-        Err(_) => get_r_version_from_r(r_paths)
+        // If the envvar is not present, then use the actual R binary to get the version.
+        Err(EnvVarError::EnvVarNotPresent) => get_r_version_from_r(r_paths),
+        // In the case of any other error than the absense of envvar, stop with
+        // that error because it means the envvar is set and something is wrong.
+        e @ Err(_) => e,
     }
 }
 
 fn set_r_version_vars(ver: &RVersionInfo) {
     println!("cargo:r_version_major={}", ver.major); // Becomes DEP_R_R_VERSION_MAJOR for clients
     println!("cargo:r_version_minor={}", ver.minor); // Becomes DEP_R_R_VERSION_MINOR for clients
     println!("cargo:r_version_patch={}", ver.patch); // Becomes DEP_R_R_VERSION_PATCH for clients
-    if ver.devel.is_empty() {
+    if ver.devel {
         println!("cargo:r_version_devel=false"); // Becomes DEP_R_R_VERSION_DEVEL for clients
     } else {
         println!("cargo:r_version_devel=true");
     }
```

Comment:
Can we replace it with one `println!("cargo:r_version_devel={}", ver.devel)`?
Also, your `if` condition seems to be inverted.

Reply:
Ouch, thanks for catching...